### PR TITLE
build(k8s.io/*): upgraded k8s libraries to v0.22.4

### PR DESCRIPTION
**Pull request by gunnar-solo**
# Description

Please include a summary of the changes.
* upgrade k8s.io/* libraries to v0.22.4
* supporting libraries necessary for upgrades bumped as well
* minor code changes to stop referencing no-longer-existing fields (that were previously depreciated)

# Context
Per https://github.com/solo-io/gloo/issues/5543, we have a desire to update our k8s libraries.  The issue _originally_ was to help resolve https://github.com/kubernetes/kubernetes/issues/91913, but, ultimately, the version upgrades are useful no matter what

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5543

### CODE REVIEW

File: projects/knative/pkg/translator/translate.go [hunk 1]
```diff
@@ -169,22 +166,6 @@ func routingConfig(ctx context.Context, ingresses map[*core.Metadata]knativev1al
 					pathRegex = ".*"
 				}
 
-				var timeout time.Duration
```
Comment (gunnar-solo) [inline projects/knative/pkg/translator/translate.go:172]: calling this out _specifically_ as a point to be reviewed.  The fields `DeprecatedRetries` and `DeprecatedTimeout` no longer exist in the required version of `knative.dev/networking`.  Following through with that upstream choice, I fully removed them.  Explanation/rationale:

> 	// DeprecatedRetries is DEPRECATED.
	// Retry in Kingress is not used anymore. See https://github.com/knative/serving/issues/6549

Comment (jenshu) [reply projects/knative/pkg/translator/translate.go:172]: that sounds reasonable to me, they don't seem to be supported/used anymore

File: go.mod [hunk 2]
```diff
@@ -136,6 +126,8 @@ replace (
 	// Required for proper serialization of CRDs
 	github.com/renstrom/dedent => github.com/lithammer/dedent v1.0.0
 
+	google.golang.org/grpc => google.golang.org/grpc v1.40.0
```
Comment (sam-heilbron) [inline go.mod:129]: Can we add a comment here outlining why we do this, or link to an issue that tracks resolving this? Just something so others have context around it / we know when we can resolve it.

File: go.mod [hunk 3]
```diff
@@ -3,15 +3,14 @@ module github.com/solo-io/gloo
...
-	github.com/Masterminds/sprig v2.22.0+incompatible // indirect
-	github.com/Microsoft/hcsshim v0.8.14 // indirect
+	github.com/Masterminds/squirrel v1.5.2 // indirect
 	github.com/Netflix/go-expect v0.0.0-20180928190340-9d1f4485533b
 	github.com/avast/retry-go v2.4.3+incompatible
 	github.com/aws/aws-sdk-go v1.34.9
+	github.com/bshuster-repo/logrus-logstash-hook v1.0.0 // indirect
 	github.com/containerd/cgroups v0.0.0-20210114181951-8a68de567b68 // indirect
-	github.com/containerd/containerd v1.4.11 // indirect
+	github.com/containerd/containerd v1.5.7 // indirect
```
Comment (sam-heilbron) [inline go.mod:13]: We use a go mod replace to pin containerd to 1.4.11, can that be removed?

**Review by sam-heilbron — COMMENTED** [review]
Small nits, but apart from that LGTM!

File: go.mod (hunk 2 above)
Comment (nfuden) [inline go.mod:129]: just for kicks whats the reason for the pin of grpc here rather than the 1.42 that is in the go mod?

Comment (gunnar-solo) [reply go.mod:131]: sure!  Just opened an issue around it:  https://github.com/solo-io/gloo/issues/5719

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**